### PR TITLE
da1469x: Switch clocks properly

### PR DESCRIPTION
**Pull request by andrzej-kaczmarek**

### CODE REVIEW

File: hw/mcu/dialog/da1469x/src/da1469x_clock.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include "syscfg/syscfg.h"
+#include "mcu/cmsis_nvic.h"
+#include "mcu/da1469x_hal.h"
+#include "mcu/da1469x_pdc.h"
+
+static inline bool
+da1469x_clock_is_32m_settled(void)
```
Comment (kasjer) [inline hw/mcu/dialog/da1469x/src/da1469x_clock.c]: it's not clear is this function is for xtal32m or something else. all other functions that operate on xtal32m are marked as such

File: hw/mcu/dialog/da1469x/src/hal_system.c [hunk 2]
```diff
@@ -128,3 +107,20 @@ hal_reset_cause(void)
...
+
+void
+da1469x_lpclk_init(void)
+{
+    static struct hal_timer lpclk_settle_tmr;
+
+    da1469x_clock_lp_xtal32k_enable();
+
+    os_cputime_timer_init(&lpclk_settle_tmr, da1469x_lpclk_settle_tmr_cb, NULL);
+    os_cputime_timer_relative(&lpclk_settle_tmr, 8000 * 1000);
```
Comment (kasjer) [inline hw/mcu/dialog/da1469x/src/hal_system.c]: Code will wait 8s for xtal32k to be ready, right?

**Comment by andrzej-kaczmarek** [comment]
@kasjer I added syscfg settings for xtal setting times

File: hw/mcu/dialog/da1469x/src/hal_system.c (hunk 2 above)
Comment (andrzej-kaczmarek) [reply hw/mcu/dialog/da1469x/src/hal_system.c]: it will not explicitly wait, it will just defer switch (lp_clk will be rc32k during this time)

**Pull request merged**